statsmodels/tsa/automatic/sarimax.py

```diff
@@ -0,0 +1,128 @@
+"""automatic forecasting for SARIMAX models."""
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+
+
+def auto_order(endog, criteria='aic', d=0, max_order=(3, 3),
+                stepwise=False, **spec):
+    """Perform automatic calculation of the parameters used in SARIMAX models.
+
+    This uses two different approaches to clacluate the parameters:
+    1. Uses a brute force approach to traverse through all the possible
+        parameters of the model.
+    2. Uses the stepwise algorithm specified in Hyndman's Paper.
+
+    Parameters
+    ----------
+    endog : list
+        input contains the time series data over a period of time.
+    criteria : str
+        specifies which information criteria to use for model evaluation.
+    d : int
+        differencing factor.
+    max_order : list
+        maximum possible values of the parameters p and q.
+    stepwise : Boolean
+        specifies whether to use the stepwise algorithm or not.
+    **spec : list of kwargs
+
+    Returns
+    -------
+    p : the autoregressive parameter for the SARIMAX model.
+    q : the moving average parameter for the SARIMAX model.
+
+    Notes
+    -----
+    Status : Work In Progress.
+    Citation : Hyndman, Rob J., and Yeasmin Khandakar.
+         "Automatic Time Series Forecasting: The forecast Package for R."
+
+    """
+    if not stepwise:
+        aic_matrix = np.zeros((max_order[0], max_order[1]))
+        for p in range(max_order[0]):
+            for q in range(max_order[1]):
+                # fit  the model
+                mod = sm.tsa.statespace.SARIMAX(endog, order=(p, d, q), **spec)
+                res = mod.fit(disp=False)
```

Comment:
You should add a try...except block here, in case something goes wrong in `fit` (for example, your unit tests are failing because of non-stationary starting parameter estimates).

Reply:
I have added the try-catch block to fit the model and I have simply raised an exception which occurs in case something goes wrong in fitting the model.